src/modules/job-info/lookup.c

```diff
@@ -234,64 +259,45 @@ static void info_lookup_continuation (flux_future_t *fall, void *arg)
     zlist_remove (ctx->lookups, l);
 }
 
-static void eventlog_lookup_continuation (flux_future_t *f, void *arg)
+/* If keys array doesn't contain eventlog, add it
+ * and update flag indicating we added it.
+ */
+static int check_keys_for_eventlog (struct lookup_ctx *l)
```

Comment:
Maybe it would simplify the code if the eventlog was not added to the json array?  Just check if it's there and if not, explicitly look it up and insert a future named `eventlog` into the composite.  Then in the continuation, first thing (if `!allow`), fetch that future from the composite by name, perform the access check, and fail right there if denied.

Then instead of iterating over the futures in the composite, iterate over the names in the json array when constructing the response.

Reply:
That one was meant as an idea that you might consider but choose to veto as more trouble than it's worth.

Reply:
after your comment above about the error string, it occurred to me that I could do something like this.  Let me see how it plays out, might be cleaner overall.